etcd/src/main/scala/io/buoyant/etcd/Key.scala

```diff
@@ -133,6 +133,38 @@ class Key(key: Path, client: Service[Request, Response]) {
   }
 
   /**
+   * Create a key in the current directory with an in-order name.
+   *
+   * If value is None, the key is treated as a directory.  In order to
+   * create an empty data node, use `Some(Buf.Empty)`.
```

Comment:
Rather than requiring this explanatory comment, could we consider using a more structured data type for this and other similar APIs in this class? something like:

```scala
sealed trait KeyType

object Directory extends KeyType
object Empty extends KeyType
case class Named(value: Buf) extends KeyType
```